### PR TITLE
🪟🧹 Improve connector form types

### PR DESCRIPTION
**Pull request by flash1293**
## What
Improve some types in relation to https://github.com/airbytehq/airbyte/issues/17352

https://github.com/airbytehq/airbyte/issues/17352 itself can be tackled once https://github.com/airbytehq/airbyte/issues/14250 is done.

## How
* All `FormItem` types inherit some properties from the underlying JSON schema, and the types should reflect that (also we need the extra `default` for the linked related task)
* Remove custom `pick` implementation and use the lodash one to reduce the number of places where typing is lost in the intermediate processing (Object.fromEntries returns an object with almost no type information)

### CODE REVIEW

File: airbyte-webapp/src/core/form/types.ts [hunk 1]
```diff
@@ -2,7 +2,16 @@ import { JSONSchema7Type, JSONSchema7TypeName } from "json-schema";
...
+/**
+ * When turning the JSON schema into `FormBlock`s,
+ * some often used props are copied over for easy access.
+ */
+type FormRelevantJSONSchema = Pick<
+  AirbyteJSONSchema,
+  "default" | "examples" | "description" | "pattern" | "order" | "const" | "title" | "airbyte_hidden" | "enum"
+>;
+
+interface FormItem extends FormRelevantJSONSchema {
```
Comment (lmossman) [inline airbyte-webapp/src/core/form/types.ts:14]: Is there a reason to keep `order`, `title`, `description`, and `airbyte_hidden` in this interface, now that they are coming from `FormRelevantJSONSchema`?

File: airbyte-webapp/src/core/jsonSchema/schemaToUiWidget.ts [hunk 2]
```diff
@@ -119,12 +121,12 @@ const defaultFields: Array<keyof AirbyteJSONSchema> = [
...
   // airbyte specific fields
   "airbyte_hidden",
-];
+] as const;
 
-const pickDefaultFields = (schema: AirbyteJSONSchema): Partial<AirbyteJSONSchema> => {
-  const partialSchema: Partial<AirbyteJSONSchema> = {
-    ...Object.fromEntries(Object.entries(schema).filter(([k]) => defaultFields.includes(k as keyof AirbyteJSONSchema))),
-  };
+type DefaultFields = Pick<AirbyteJSONSchema, typeof defaultFields[number] | "enum">;
```
Comment (lmossman) [inline airbyte-webapp/src/core/jsonSchema/schemaToUiWidget.ts:126]: A couple questions here:
1. What is `typeof defaultFields[number]` doing? It seems like this is just trying to pick all of the fields in the `defaultFields` array from `AirbyteJSONSchema`, but I don't know what that `number` in the brackets is being defined. Is this some functionality specific to `Pick`?
2. Why not just add `"enum"` to the `defaultFields` array? This way you wouldn't need to do ` | "enum"` unless I've missed something

**Review by lmossman — COMMENTED** [review]
Had a couple questions

File: airbyte-webapp/src/core/form/types.ts (hunk 1 above)
Comment (flash1293) [reply airbyte-webapp/src/core/form/types.ts:14]: There is not, good catch.

**Comment by flash1293** [comment]
Based on your comments I simplified a bit further, let me know if there are other questions.

**Review by lmossman — APPROVED** [review]
Changes LGTM. Tested with connectors that have conditional fields and didn't notice any regression

**Pull request merged**